src/Serilog.Sinks.AzureDocumentDb/Sinks/AzureDocumentDb/AzureDocumentDbSink.cs

```diff
@@ -35,42 +36,37 @@ public AzureDocumentDBSink(Uri endpointUri, string authorizationKey, string data
             _formatProvider = formatProvider;
             _client = new DocumentClient(endpointUri, authorizationKey);
             _storeTimestampInUtc = storeTimestampInUtc;
-            Task.WaitAll(new []{CreateDatabaseIfNotExistsAsync(databaseName)});
-            Task.WaitAll(new []{CreateCollectionIfNotExistsAsync(collectionName)});
+            CreateDatabaseIfNotExistsAsync(databaseName).ConfigureAwait(false);
+            CreateCollectionIfNotExistsAsync(collectionName).ConfigureAwait(false);
         }
 
         public void Emit(LogEvent logEvent)
         {
-            Task.WaitAll(EmitAsync(logEvent));
+            EmitAsync(logEvent).ConfigureAwait(false);
```

Comment:
.Wait() is needed here too, I think (though the sink looks like it would benefit from subclassing `PeriodicBatchingSink` and thus batching up these writes).